model/task/task_annotations.go

```diff
@@ -195,8 +199,12 @@ func PatchAnnotation(a *annotations.TaskAnnotation, userDisplayName string, upse
 	); err != nil {
 		return errors.Wrapf(err, "updating task annotation for '%s'", a.TaskId)
 	}
-	if a.Issues != nil && len(a.Issues) > 0 {
-		return SetHasAnnotations(a.TaskId, a.TaskExecution)
+	if a.Issues != nil {
+		if len(a.Issues) == 0 {
+			return UnsetHasAnnotations(a.TaskId, a.TaskExecution)
```

Comment:
if this is too big of a refactor for this PR we can split a ticket, but i feel like we should just have one function, that sets annotations to true or false based on the length of issues (or at least that takes in a boolean, so we could just pass through len(a.Issues)==0 as the bool), rather than having this if statement

```
hasAnnotations := len(a.Issues) > 0
return UpdateHasAnnotations(a.TaskId, a.TaskExecution, hasAnnotations)

Reply:
you got it boss ‼️